frontend/src/components/Share/Share.js

```diff
@@ -1,100 +1,59 @@
 
-class Share extends React.Component {
-  constructor() {
-    super();
-
-    this.handleClick = this.handleClick.bind(this);
-    this.handleOutsideClick = this.handleOutsideClick.bind(this);
-
-    this.state = {
-      showModal: false,
-      copySuccess: false,
-
-    };
-  }
-
-  handleClick() {
-    if (!this.state.showModal) {
-      // attach/remove event handler
-      document.addEventListener('click', this.handleOutsideClick, false);
-    } else {
-      document.removeEventListener('click', this.handleOutsideClick, false);
-    }
-
-    this.setState(prevState => ({
-       showModal: !prevState.showModal,
-    }));
-  }
-
-  handleOutsideClick(e) {
-    // ignore clicks on the component itself
-    if (this.node.contains(e.target)) {
-      return;
-    }
-
-    this.handleClick();
-  }
-
-   copyCodeToClipboard = () => {
-    this.urlRef.select()
-    document.execCommand("copy")
-    this.setState({copySuccess: true})
-      setTimeout(() => {
-          this.setState(prevState => ({
-       copySuccess: !prevState.copySuccess,
-       showModal: !prevState.showModal,
-    }))
-         }, 3000);
-
+function Share() {
+
+  const [modalShow, setModalShow] = useState(false);
+  const [copySuccess, setcopySuccess] = useState(false);
+  const message = "Share Covexit with others and help during this crisis.";
+  const inputRef = useRef(null);
+  const urlRef = window.location.href;
+
+  function copyLink() {
+    inputRef.current.select();
+    document.execCommand("copy");
+    setcopySuccess("∞ copied!");
+    setTimeout(() => {
+      setModalShow(false);
+      setcopySuccess(false);
+    }, 3000);
   }
 
-  render() {
-    return (
-      <div className="Share" ref={node => { this.node = node; }}>
-        <button className="Share-toggle"
-          onClick={this.handleClick}
-        ><img src={share} alt="Share Icon" className="Share-icon" />
-         Share
-        </button>
-        {this.state.showModal && (
-          <div className="modal">
-          <div className ="message">Share Covexit with others and help during this crisis.
-          </div>
-
-          <input id="copy"
-
-            ref={(urlRef) => this.urlRef = urlRef}
-            value={window.location.href}
-
-          />
-
-            {
-          this.state.copySuccess ?
-            <div id="copy">
-           ∞ copied!
-            </div> : null
-
-           }
+  return (
+    <div className="Share">
+      <button className="Share-toggle" onClick={() => setModalShow(!modalShow)}>
+        <img src={share} alt="Share Icon" className="Share-icon" />
+        Share
+      </button>
+
+      <div className={`Modal Modal--${modalShow ? "opened" : "closed"}`}>
+        <div className="Modal-body">
+          <div className="Message">{message}</div>
+          <input className="Copy-Link" ref={inputRef} value={urlRef} />
+
+          <div class="Success">{copySuccess}</div>
+
+          <div className="Btn-group">
+            <button
```

Comment:
No, I want you to use the button component. Like so: https://github.com/Covexit/covexit/blob/master/frontend/src/views/Home.js#L18